src/common/transformations/src/transformations/common_optimizations/fq_reshape_fusion.cpp

```diff
@@ -62,21 +66,41 @@ ov::pass::FakeQuantizeReshapeFusion::FakeQuantizeReshapeFusion() {
                                });
                 const auto& new_limit_size = shape_size(new_limit_shape);
                 if (new_limit_size == limit_size) {  // we tracked future channel placement
-                    if (new_limit_shape == limit_input.get_shape())
+                    if (new_limit_shape == limit_input.get_shape()) {
                         renewed_inputs.push_back(limit_input);
-                    else
-                        renewed_inputs.push_back(reshape_node->clone_with_new_inputs(
+                    } else {
+                        auto reshaped_input = reshape_node->clone_with_new_inputs(
                             {limit_input,
-                             ov::op::v0::Constant::create(element::i64, {new_limit_shape.size()}, new_limit_shape)}));
+                             ov::op::v0::Constant::create(element::i64, {new_limit_shape.size()}, new_limit_shape)});
+                        if (auto constant = ov::util::get_constant_from_source(reshaped_input)) {
+                            reshaped_input = constant;
+                        }
+                        renewed_inputs.push_back(reshaped_input);
```

Comment:
We can do that, sure. The only difference is that clone_try_fold only works for single node folding and get_constant_from_source checks all the necessary no_folding rt info and could fold a chain of nodes. For now, I won't change the code in the PR, however, in case more feedback arises -- I'll clean it up.